node/pkg/watchers/evm/watcher.go

```diff
@@ -595,36 +608,42 @@ func (w *Watcher) Run(parentCtx context.Context) error {
 				logger.Debug("processing new header",
 					zap.Stringer("current_block", ev.Number),
 					zap.Stringer("current_blockhash", currentHash),
-					zap.Bool("is_safe_block", ev.Safe),
+					zap.Stringer("finality", ev.Finality),
 					zap.String("eth_network", w.networkName))
 				readiness.SetReady(w.readinessSync)
 
-				w.pendingMu.Lock()
-
 				blockNumberU := ev.Number.Uint64()
-				if ev.Safe {
-					atomic.StoreUint64(&currentSafeBlockNumber, blockNumberU)
+				if ev.Finality == connectors.Latest {
+					atomic.StoreUint64(&w.latestBlockNumber, blockNumberU)
+					currentEthHeight.WithLabelValues(w.networkName).Set(float64(blockNumberU))
+					w.updateRegistry()
+					continue
+				}
+
+				// The only blocks that get here are safe and finalized.
+
+				if ev.Finality == connectors.Safe {
+					atomic.StoreUint64(&w.latestSafeBlockNumber, blockNumberU)
+					currentEthSafeHeight.WithLabelValues(w.networkName).Set(float64(blockNumberU))
 				} else {
-					p2p.DefaultRegistry.SetNetworkStats(w.chainID, &gossipv1.Heartbeat_Network{
-						Height:          ev.Number.Int64(),
-						ContractAddress: w.contract.Hex(),
-					})
 					atomic.StoreUint64(&currentBlockNumber, blockNumberU)
 					atomic.StoreUint64(&w.latestFinalizedBlockNumber, blockNumberU)
-					currentEthHeight.WithLabelValues(w.networkName).Set(float64(ev.Number.Int64()))
+					currentEthFinalizedHeight.WithLabelValues(w.networkName).Set(float64(blockNumberU))
```

Comment:
Why a float64?  Is that just what gauges take?

Reply:
Yep https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#Gauge